lib/vector-core/src/transform/mod.rs

```diff
@@ -9,20 +9,20 @@ pub mod runtime_transform;
 ///
 /// While function transforms can be run out of order, or concurrently, task
 /// transforms act as a coordination or barrier point.
-pub enum Transform {
-    Function(Box<dyn FunctionTransform>),
-    Task(Box<dyn TaskTransform>),
+pub enum Transform<T: Clone + Send + 'static> {
+    Function(Box<dyn FunctionTransform<T>>),
+    Task(Box<dyn TaskTransform<T>>),
 }
 
-impl Transform {
+impl<T: Clone + Send> Transform<T> {
```

Comment:
I did that initially, but found that it would make sense to reuse it based on your suggestion: https://github.com/timberio/vector/pull/7482/commits/cee58ad230f51ec149580039f77545fc7f157e41#r641802326

> This is effectively the same as a `TaskTransform`, which implies to me that something is off here. We shouldn't have two different concepts with exactly the same function signature.

I might have interpreted that too broadly, but I found that the solution to the framing problem can be solved in the exact same way.